### ISSUE TITLE
'Add to tick list' / wishlist / target vs normal ticking / (aka circuits)

### ISSUE THREAD

**Issue by brendanheywood** [theCrag/website#1318]
Had a climb with Gene from the blueys today at Fruehoff and his no 1 desired feature was proper targets / tick lists. I hear this a bit, anyway we talked about a few ideas and documenting them here:

The usual suspects:
- [ ] Add to tick list in list view, single click and done in ajax
  ![image](https://f.cloud.github.com/assets/187449/1877493/378a8870-7928-11e3-9af6-c0ceb40c0f1a.png)
- [ ] Add to tick list in route page, should be a single click ajax like fav's
- [ ] sort out all the filtering so they don't mess up the various stats
- [ ] But the real cool feature would be to see a new page just for your tick list, either sort by grade, or sort by when it was added, but then next to each targeted tick, have a sort of little history of attempts at that route. It could be sorted by the best tick type of the attempts you've had, so its easy to see the progression of how a person has set a bunch of goals and where they are up to in achieving them. I think it would be awesome to see JJ's or Lee's tick history laid out like this. In fact I'd probably go one step further and if someone has tried a route more than say 3 times it gets onto the ticks list without the explicit target.

* [ ] a list can be positioned at a node (automatic like topos, but can be overridden)
* [ ] be able to like / star a list from someone else
* [ ] have a comments thread under it
* [ ] be able to see the list and your ticks against it, and also the ticks of the person who made it for direct comparison

Bonus points:
* [ ] custom comment for each addition to the list, eg why this was added to the list
* [ ] the list should have the option to either hide / auto remove items when you tick them

Super ideas:
* [ ] be able to 'join' a list, which puts you down as a contender for that list. When you view the list you see a matrix of all contenders, but ranked by who has most ticked the list

**Comment by guyfi** [comment]
yes would love this feature. 
happy to test it out if you can allow it for beta to certain users.

**Comment by brendanheywood** [comment]
Cris had an idea which should fit in with this of everyone having a 'top ten' list which affects popularity score. Thinking that any route added to one or more list gets counted towards the score but only once. Visibility from profile page was always the intention for the new lists

**Comment by DaneEvans** [comment]
I'd like to see something happen with this, Brendan's implementation looks good for my purposes. 

To be honest the existing circuits implementation would work if it weren't for the fact that I can't create another one ... I don't want a list of Araps easy solos in with my list of Canberra and surrounds harder stuff that I'd like to get on when I have a day or two free. 

My main concerns would be 
- the ability to have multiple lists
- view the routes within them easily 
- ability to make them named and public
- ability to attach to areas

If the add to wishlist button, and ability to see other people's lists is easier, I think it would be well used and received.

**Comment by scd** [comment]
We are bogged down in a couple of structural issues. I think that not being able to create a circuit is a regression issue. see issue #2378

**Comment by scd** [comment]
I have put back a link. @daneevans can you try and create a circuit using the action button. Navigate to the area you want first.

**Comment by DaneEvans** [comment]
It's working, but the feed info needs updating
![image](https://cloud.githubusercontent.com/assets/22091505/18435415/6f10b70e-7936-11e6-8958-b2377e187cdd.png)

**Comment by brendanheywood** [comment]
I'm going to close https://github.com/theCrag/website/issues/1528 and merge it into here

**Comment by quaestor** [comment]
Anything new on this front? I have a few ideas for circuits, but with the current functionality, it's probably not worth collecting the data if it's unclear the circuits will be merged into the overhauled ticklists.

One feature I would like to see in the circuits feature would be custom comments for the routes, so for example you can explain *why* you added a route to the list.

**Comment by scd** [comment]
No movement on this one yet, but it is still important. Good idea bout
custom comment.

**Comment by brendanheywood** [comment]
@quaestor this is the number 3 most requested feature, so priority wise for us it near the top (behind improvements to the maps and forums). So we can't guarantee timelines but it's getting closer. Under the hood the circuit data model is quite good, most of this is just a UX revamp exercise, and then after that a migration of the existing target data into a personal list for each person.

If you have ideas I'd love to hear them earlier so it helps informs our roadmap. Either in this issue, and / or if what you are wanting is more around the Fontainbleau concept of circuits then this issue might be more appropriate: https://github.com/theCrag/website/issues/2639

**Comment by quaestor** [comment]
Okay, I'll sum up what I think:

* major:
- [ ] possibility to reorder routes in the list
- [ ] separate "detail" page
- [ ] custom comment for each addition to the list
* minor:
- [ ] other users should be able to "suggest" routes
- [ ] instead of only routes, allow arbitrary nodes, to allow lists like "best crags in area XY"
* negligible:
- [ ] to reduce clutter, initially allow only one list per user (personal ticklist) and scale the number of allowed lists with crag karma
- [ ] similar to the "suggestions" feature, one could add a voting system, like the current "starred by" feature for circuits, but on the route level

A "circuit" I've recently created is "First Female Ascents in Frankenjura". I added the few routes I know about from references, but the first point in the "minor" section would be a good way for the community to contribute to this list; the points of "negligible" could add to the "social" aspect. A list I wanted to create is "proper summits in the Frankenjura", for which without my second "minor" point I would have to choose specific routes at the relevant crags.

**Comment by brendanheywood** [comment]
I really like your list, in particular the idea of a custom comment for why it's on the list :)

Can you also flesh out what you want on the detail page? What I'm envisaging here is the list of the routes in order, and your ticks next to all of them. If you are looking at someone else's public tick list you can see your ticks and their ticks side by side to compare.

**Comment by quaestor** [comment]
That depends of course on what features will be implemented. +1 for the tick list comparison, that would be awesome. I imagine the detail page of tick lists somewhat like regular node pages. The list of routes (showing the custom comment instead of the route beta description, and "+"/"-" buttons appearing while hovering instead of the "edit route" button, for the voting feature) plus the description above, maybe even a webcover; for the suggestions feature, something like a "route picker".

**Comment by brendanheywood** [comment]
So I had a mate chat to me tonight at the gym about this great idea ... which is exactly this circuits / wishlist idea :)

Another thing dropped out of the discussion:  We already allow annotations in between routes, and we should have a similar concept between routes in a list / circuit. This might also kill 2 birds and be a good enough place to add extra arbitrary stuff like why the following route(s) was added (see @quaestor idea above). Also if the list is a circuit at a particular crag, the annotations could also optionally be located and act as additional waypoints

**Comment by lordyavin** [comment]
I have a question/problem that may fit here. I tried to search for routes in a specific area that I've never climbed and wanted to include those I still lag the redpoint ascent. Ther result may act as a todo/tick list for that area and it would be great to print that list without any site navigation stuff.

**Comment by brendanheywood** [comment]
@lordyavin you can almost get this with the current filters and sorting, but there was a few bugs which I have just ironed out.

https://github.com/theCrag/website/issues/3159

After the next release you can sort the routes by tick type which will show your red points at the top and your attempts at the bottom. Note this one is in the new design release which will probably be 2 releases away.

I also really like this idea in the context of the todo lists so they will be aligned when we get to them

**Comment by lordyavin** [comment]
You didn't address the print version of the search result. Any ideas on that?

**Comment by brendanheywood** [comment]
The print stuff is kinda tangential and should probably be something generic across the board so any page gets a more stream lined print. I've logged that here:

https://github.com/theCrag/website/issues/3165

**Comment by brendanheywood** [comment]
+1 from facebook support query

**Comment by brendanheywood** [comment]
+1 via https://github.com/theCrag/website/issues/3288 and https://github.com/theCrag/website/issues/3287

**Comment by oni303** [comment]
I was searching for a way to group routes independent from there location/original grouping.
The goal was to  add boulder circuits to an area. 

Features I would need for that are:
- assign circuit to an area
- see circuits assigned to the area on its side (not sure whether circuits of sub-areas should be included) 
- add routes of different sub-areas
- order routes in the list as needed

Tick-lists are another good idea, may be it is better to realize them separately, as there are some features which are not wanted in the other one e.g.:
- checking a route publicly, boulder circuits should stay unmodified
- suggesting routes, may be not bad but certainly used very rarely

In my opinion it makes the usage more easy if these are two separate features.

**Comment by brendanheywood** [comment]
+1 from new supporter

**Comment by lordyavin** [comment]
> 
> 
> @lordyavin you can almost get this with the current filters and sorting, but there was a few bugs which I have just ironed out.
> 
> #3159
> 
> After the next release you can sort the routes by tick type which will show your red points at the top and your attempts at the bottom. Note this one is in the new design release which will probably be 2 releases away.
> 
> I also really like this idea in the context of the todo lists so they will be aligned when we get to them

I tried to assemble a query that match my "not red pointed routes". No luck so far. could you give me a hint which settings I have to combine?

**Comment by rouletout** [comment]
+1 for creating / re-introducing wish-lists for trips and in general

**Comment by simrun** [comment]
+1 I want to plan routes for my next trip to the crag

**Comment by brendanheywood** [comment]
In the mean time the workaround is just adding 'target' ticks, log them like a normal tick. Then you can filter on your target ticks to see your todo list

**Comment by rouletout** [comment]
+1 from FB (https://www.facebook.com/groups/189656637772644/permalink/2564834223588195/)

**Comment by brendanheywood** [comment]
+1 from new supporter

**Comment by brendanheywood** [comment]
From @dmucli via https://github.com/theCrag/website/issues/3417

I've also had 1 or 2 other people also make suggestions of allowing crags in the lists as well as routes. I don't see any reason to restrict it. Some wish lists could be a mix of specific boulder problems and a whole boulder. 

---

A feature that lets (some?) users create lists of areas or routes featured at the world level, continent level or country level.

**Just a few examples**

At the world level 
World: *Most scenic crags in the world (Yosemite Valley, Wadi Rum, Getu, Verdon Gorges...)*
World: *Hardest multi-pitches routes (Dawn Wall, Silbergeier, Orbayu, Tough Enough etc.)*

At the continent level
North America: *Best sport climbing crags* (Red River Gorge, The Gunks, Red Rocks etc)*
Europe: *Climbing above the sea (Mallorca, Les Calanques, Kalymnos, Sardinia...)*
Europe: *Best bouldering areas (Fontainebleau, Magic Wood, Albarracin...)*
Asia: *Best DWS destinations (Halong Bay, Tonsai, El Nido)*

At the country level
Spain: *Best 8a sport routes* or *top 20 best sport routes under 8a*
France: *Routes at high altitudes*

The list goes on! 

This could be a cool feature to help climbers discover areas/routes in places they want to travel to. It could also be a more valuable way of presenting classic routes in a country instead of the "nearby icons" feature which is limited to one route per grade. 

Extending on the ideas of lists, there could also be lists not related to any area and that can be created by any users (privately or publicly) such as "my favourite routes of all time", "routes I want to climb before I die" etc. But this is another topic.

**Comment by brendanheywood** [comment]
+1 from support nick

**Comment by brendanheywood** [comment]
+1 from supporter blake

**Comment by brendanheywood** [comment]
+1 support email

**Comment by brendanheywood** [comment]
+1 supporter

**Comment by brendanheywood** [comment]
+1 supporter

**Comment by Mdemaillard** [comment]
+10 or so from forum https://www.thecrag.com/discussion/3217181775/the-possibility-of-a-goals-tick-list-on-thecrag

**Comment by nicHoch** [comment]
Hi Juergen

you can search for all closed area or warning in Frankenjura:

Warnings created by igkletternfrankenjura

https://www.thecrag.com/en/climbing/germany/frankenjura-nord/discussions/from/igkletternfrankenjura/of-discussion-type/warning/?sortby=when,desc

Closed area:

https://www.thecrag.com/en/climbing/germany/frankenjura-nord/areas/with-structured-tag/198053209/?sortby=at,desc&page=1

you could add this links in the Frankenjura description as a connivance shortcut. 
But anyway we are working right now on the "list" function - that should be ready around summer. But i am not sure if lists are a good way to handle closings as the search function will be always up to date.

**Comment by rouletout** [comment]
Most of this is available with the new list feature introduced in release 92. Closing.
